src/omp/kernels/distributed.f90

```diff
@@ -227,4 +227,104 @@ subroutine der_univ_subs(du, recv_u_s, recv_u_e, n, dist_sa, dist_sc)
 
   end subroutine der_univ_subs
 
+  subroutine der_univ_fused_subs(rhs_du, dud, d2u, v, &
+                                 du_recv_s, du_recv_e, &
+                                 dud_recv_s, dud_recv_e, &
+                                 d2u_recv_s, d2u_recv_e, &
+                                 nu, n, du_dist_sa, du_dist_sc, &
+                                 dud_dist_sa, dud_dist_sc, &
+                                 d2u_dist_sa, d2u_dist_sc)
+    implicit none
+
+    ! Arguments
+    real(dp), intent(inout), dimension(:, :) :: rhs_du
+    real(dp), intent(in), dimension(:, :) :: dud, d2u, v
+    real(dp), intent(in), dimension(:, :) :: du_recv_s, du_recv_e
+    real(dp), intent(in), dimension(:, :) :: dud_recv_s, dud_recv_e
+    real(dp), intent(in), dimension(:, :) :: d2u_recv_s, d2u_recv_e
+    real(dp), intent(in), dimension(:) :: du_dist_sa, du_dist_sc
+    real(dp), intent(in), dimension(:) :: dud_dist_sa, dud_dist_sc
+    real(dp), intent(in), dimension(:) :: d2u_dist_sa, d2u_dist_sc
+    real(dp), intent(in) :: nu
+    integer, intent(in) :: n
+
+    ! Local variables
+    integer :: i, j
+    real(dp) :: ur, bl, recp
+    real(dp), dimension(SZ) :: du_s, du_e, dud_s, dud_e, d2u_s, d2u_e, &
+                               temp_du, temp_dud, temp_d2u
+
+    !$omp simd
+    do i = 1, SZ
+      ! A small trick we do here is valid for symmetric Toeplitz matrices.
+      ! In our case our matrices satisfy this criteria in the (5:n-4) region
+      ! and as long as a rank has around at least 20 entries the assumptions
+      ! we make here are perfectly valid.
```

Comment:
would not complying with the assumption lead to wrong results or poor performance?
If it is wrong results it would probably be good to actively check for this somewhere (probably higher up in the call stack) and throw an error if the assumptions aren't met.

Reply:
It would lead to wrong results. Created an issue yo deal with this in a dedicated PR later. #133